File: app/src/main/java/com/example/groxweatherapp/view/WeatherMainView.java
package com.example.groxweatherapp.view;

import com.example.groxweatherapp.model.Weather;
import java.util.List;

public interface WeatherMainView {

    void showSpinner();

    void hideSpinner();

    void showList();

    void hideList();

    void showWeather(List<Weather> weatherList);

    void showErrorDialog();
}

File: app/src/main/java/com/example/groxweatherapp/view/WeatherMainView.kt
package com.example.groxweatherapp.view

import com.example.groxweatherapp.model.Weather

interface WeatherMainView {

    fun showSpinner()

    fun hideSpinner()

    fun showList()

    fun hideList()

    fun showWeather(weatherList: List<Weather>)

    fun showErrorDialog()
}
